### PR TITLE
DOCSP-13850 indexes

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
# Pull Request Info

Added the Fundamentals > Indexes page -- this also documents clustered indexes (I closed this ticket previously since the tasks overlapped).

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-13850>
Staging - <https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-13850/fundamentals/indexes/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/includes/fundamentals/code-snippets/indexes.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+		// begin milti key index
+		indexModel := mongo.IndexModel{Keys: bson.D{{"cast", -1}}}
+		name, err := coll.Indexes().CreateOne(context.TODO(), indexModel)
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Println("Name of Index Created: " + name)
+		// end milti key index
```
Comment (cbush) [inline source/includes/fundamentals/code-snippets/indexes.go:113]: multi?

File: source/includes/fundamentals/code-snippets/indexes.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Println("Name of Index Created: " + name)
+		// end geo index
+
+		res, err := coll.Indexes().DropOne(context.TODO(), "location.geo_2dsphere")
+
+		// res, err := coll.Indexes().DropOne(context.TODO(), "geo index")
```
Comment (qingyang-hu) [inline source/includes/fundamentals/code-snippets/indexes.go:161]: Do we need to keep this line in the code?

**Review by matthewdale — APPROVED** [review]
Looks good 👍

**Review by qingyang-hu — APPROVED** [review]
LGTM, thank you!

**Pull request merged**